utility.lua

```diff
@@ -129,25 +129,39 @@ hopper.take_item_from = function(hopper_pos, target_pos, target_node, target_inv
 	end
 end
 
-local function send_item_to_air(hopper_inv, target_pos, filtered_items)
+local function send_item_to_air(hopper_inv, target_pos)
+	local item
 	local stack
 	local stack_num
 	for i = 1, hopper_inv:get_size("main") do
 		stack = hopper_inv:get_stack("main", i)
-		local item = stack:get_name()
-		if item ~= "" and (filtered_items == nil or filtered_items[item]) then
+		item = stack:get_name()
+		if item ~= "" then
 			stack_num = i
 			break
 		end
 	end
-	if not stack_num then
-		return false
+
+	local eject_node = minetest.get_node(target_pos)
+	if (eject_node["name"] ~= "air") then
+		minetest.log("verbose", "hopper.send_item_to_air: eject direction not air ("..eject_node["name"].." at "..target_pos:to_string().."). Looking for alternate.")
+		local air_pos
+		local radius = 1
+		while not air_pos and radius <= 5 do
+			air_pos = minetest.find_node_near(target_pos, radius, {"air"})
+			radius = radius + 1
+		end
```

Comment:
Inefficient. Plenty of nodes will be checked multiple times. Checking with radius 2 once should be enough. Indicate failure by return value.

Reply:
Right... newbie me. Over-complicated for no good reason.